datalake/src/main/java/com/sequenceiq/datalake/service/sdx/status/SdxStatusService.java

```diff
@@ -53,11 +58,46 @@
     @Inject
     private Clock clock;
 
+    @Inject
+    private FlowLogService flowLogService;
+
+    @Inject
+    private FlowChainLogService flowChainLogService;
+
     public void setStatusForDatalakeAndNotify(DatalakeStatusEnum status, String statusReason, SdxCluster sdxCluster) {
         setStatusForDatalake(status, statusReason, sdxCluster);
         sdxNotificationService.send(status.getDefaultResourceEvent(), sdxCluster);
     }
 
+    public void setStatusForDatalakeAndNotify(DatalakeStatusEnum status, String statusReason, SdxCluster sdxCluster, long flowId) {
+        FlowLog flowLog = flowLogService.getLastFlowLog(flowId).get();
+        setStatusForDatalake(status, statusReason, sdxCluster);
+        FlowChainLog flowChainLog = flowChainLogService.findFirstByFlowChainIdOrderByCreatedDesc(flowLog.getFlowChainId()).get();
+        if (!(flowChainLog.getFlowChainType().equals(DatalakeResizeFlowEventChainFactory.class.getName()) &&
+                status.isFailedState())) {
+            sdxNotificationService.send(status.getDefaultResourceEvent(), sdxCluster);
+        }
+    }
+
+    public SdxCluster setStatusForDatalakeAndNotify(DatalakeStatusEnum status, String statusReason, Long datalakeId, String flowId) {
+        FlowLog flowLog = flowLogService.getLastFlowLog(flowId).get();
+        FlowChainLog flowChainLog = flowChainLogService.findFirstByFlowChainIdOrderByCreatedDesc(flowLog.getFlowChainId()).orElse(null);
```

Comment:
I might like this pattern a bit more if the flow chain service was doing the lookup for the flow log.
@Nullable 
FlowChainLog flowChainLog = flowChainLogService.findParentChain(flowId)